README.md
# Car Lease Calculator :red_car:

[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)
[![Npm Version](https://img.shields.io/npm/v/lease-calculator?color=blue&logo=npm-version)](https://www.npmjs.com/package/lease-calculator)
![](https://github.com/ErezNagar/lease-calculator/workflows/Tests/badge.svg)

A simple auto lease calculator for calculating your monthly lease payment, APR, total lease cost, percentage off MSRP, etc.

See it used in the Auto Lease app: https://github.com/ErezNagar/lease-calculator-app

## How To Use

```bash
npm install lease-calculator --save
```

```javascript
const leaseCalculator = new LeaseCalculator();
leaseCalculator.calculate({
  // Make of the vehicle, for calculating manufacturer-based fees
  make: "Toyota",
  // Required, MSRP of the vehicle
  msrp: 23000,
  // Required, negotiated price of the vehicle
  sellingPrice: 21000,
  // Required, Residual value of the vehicle
  rv: 13110,
  // Whether the RV is an absolute value or a percentage of MSRP. Default: true
  isRVPercent: false,
  // Required, The money factor of the lease
  mf: 0.00125,
  // The state's sales tax in percentage. Default: 0%.
  salesTax: 6.25,
  // Total fees of the lease. Default: 0.
  totalFees: 1200,
});

// Get the lease monthly payment
const monthlyPayment = leaseCalculator.getMonthlyPayment();
```

## API

|               | Description                                                                                                                                      |
| ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
| `calculate()` | Main function to calculate lease total & monthly payments, APR, total cost, etc. Must be run first to calculated the different lease parameters. |

Arguments:

```javascript
calculate({
  // Make of the vehicle, for calculating manufacturer-based fees. Default: "".
  // See constants.js for valid values.
  make: "Toyota",
  // Required, MSRP of the vehicle
  msrp: 23000,
  // Required, negotiated price of the vehicle
  sellingPrice: 21000,
  // Required, Residual value of the vehicle.
  // If isRVPercent is true, value must be a percentage. I.e, if RV is 65%, rv should be 65.
  rv: 13110,
  // Whether the RV is an absolute value or a percentage of MSRP. Default: true
  isRVPercent: false,
  // Required, The money factor of the lease
  mf: 0.00125,
  // The length of the lease in months. Default: 36.
  leaseTerm: 36,
  // The state's sales tax in percentage. Default: 0%.
  salesTax: 6.25,
  // Total fees of the lease. Default: 0.
  totalFees: 1200,
  // Total discount from dealer and manufacturer. Default: 0.
  rebates: 500,
  // Total trade-in value. Default: 0.
  tradeIn: 0,
  // Down payment, if applicable. Default: 0.
  downPayment: 0,
  // Method of taxation to apply, based on state. Default: TAX_ON_MONTHLY_PAYMENT
  // See constants.js for valid values.
  taxMethod: 1,
});
```

|                                       |                                                                                                                                                                                    |
| ------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `getRVValue()`                        | Gets the residual value of the lease                                                                                                                                               | Number |
| `getRVPercentage()`                   | Gets the residual value of the lease in percentage                                                                                                                                 | Number |
| `getMonthlyPaymentPreTax()`           | Gets the monthly payment of the lease, not including taxes                                                                                                                         | Number |
| `getMonthlyPayment()`                 | Gets the monthly payment of the lease, including taxes                                                                                                                             | Number |
| `getDiscountOffMsrpPercentage()`      | Gets the discount off of MSRP, in percentage                                                                                                                                       | Number |
| `getMonthlyPaymentToMsrpPercentage()` | Gets the percentage of the monthly payment out of the MSRP                                                                                                                         | Number |
| `getTotalLeaseCost()`                 | Gets the total cost of the lease. This includes all monthly payments, down paymenaand dealer fees. Does not include disposition fee, vehicle maintenance cost or other cost of use | Number |
| `getAPR()`                            | Gets the APR value of the lease                                                                                                                                                    | Number |
| `getAcquisitionFeeValue`              | Gets the acquisition fee value by brand. If no brand sepcified, returns 0                                                                                                          | Number |
| `getDispositionFeeValue`              | Gets the disposition fee value by brand. If no brand sepcified, returns 0                                                                                                          | Number |
| `getDriveOffPayment`                  | Gets total drive-off payment                                                                                                                                                       | Number |

## Supported manufacturers

Lease Calculator supports acquisition and disposition fee calculation for the following brands:

Acura, Alfa Romeo, Audi, Bmw, Buick, Cadillac, Chevrolet, Chrysler, Dodge, Fiat, Ford, Genesis, GMC, Honda, Hyundai, Infiniti, Jaguar, Jeep, Kia, land rover, Lexus, Lincoln, Mini, Mazda, Mercedes benz, Mitsubishi, Nissan, Ram, Scion, Smart, Subaru, Tesla, Toyota, Volkswagen, Volvo


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/).

## [1.2.0](https://github.com/ErezNagar/lease-calculator/compare/v1.1.0...v1.2.0) - 27/7/2020

### Added

 - Support for different taxation methods:
    Tax levied on monthly payment (most states)
    Tax levied on selling/sales price (Virginia)
    Tax levied on total lease ppayment (New York, New Jersy, Minnesota, Ohio, Georgia)

## [1.1.0](https://github.com/ErezNagar/lease-calculator/compare/v1.0.1...v1.1.0) - 29/6/2020

### Added

- Acquisition and disposition fee support for 35 manufacturers:
    calculate() now accepts a `make` field for calculating manufacturer-based fees.
- getAcquisitionFeeValue(), returns acquisition fee value
- getDispositionFeeValue(), returns disposition fee value

## [1.0.1](https://github.com/ErezNagar/lease-calculator/compare/v1.0.0...v1.0.1) - 27/4/2020

### Fixed

- Removed required support for the experimental classProperties syntax
- Fixed package.json/main property

## [1.0.0](https://github.com/ErezNagar/lease-calculator/compare/v0.1.0...v1.0.0) - 27/4/2020

- Version official version

## [0.1.0](https://github.com/ErezNagar/lease-calculator/releases/tag/v0.1.0) - 14/4/2020

Initial version


tests/constants.js
import {
  TAX_ON_SALES_PRICE,
  TAX_ON_TOTAL_LEASE_PAYMENT,
} from "../src/constants";

export const DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT = {
  make: "Toyota",
  msrp: 23000,
  sellingPrice: 21000,
  rv: 13110,
  isRVPercent: false,
  mf: 0.00125,
  leaseTerm: 36,
  salesTax: 10.25,
  totalFees: 1200,
  rebates: 500,
};

export const DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT = {
  make: "Toyota",
  msrp: 23000,
  sellingPrice: 21000,
  rv: 13110,
  isRVPercent: false,
  mf: 0.00125,
  leaseTerm: 36,
  salesTax: 10.25,
  totalFees: 1200,
  rebates: 500,
  rebates: 0,
  downPayment: 1700,
};

export const DUMMY_LEASE_WITH_PERCENTAGE_RV = {
  ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
  rv: 57,
  isRVPercent: true,
};

export const RV_VALUE = 13110;
export const RV_PERCENTAGE = 57;
export const ACQUISITION_FEE_TOYOTA = 650;
export const DISPOSITION_FEE_TOYOTA = 350;

export const WHEN_TAX_ON_MONTHLY_PAYMENT = {
  APR: 3,
  OFF_MSRP: 8.7,
  PAYMENT_ZERO_DOWN: 331.84,
  PAYMENT_WITH_DOWN: 293.44,
  MSRP_PERCENTAGE: 1.44,
  PAYMENT_ZERO_DOWN_PRE_TAX: 300.99,
  TOTAL_LEASE_COST_ZERO_DOWN: 13619.36,
  TOTAL_LEASE_COST_WITH_DOWN: 14111.07,
};

export const DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_SALES_PRICE = {
  ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
  taxMethod: TAX_ON_SALES_PRICE,
};

export const DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_SALES_PRICE = {
  ...DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
  taxMethod: TAX_ON_SALES_PRICE,
};

export const WHEN_TAX_ON_SALES_PRICE = {
  PAYMENT_ZERO_DOWN: 300.99,
  PAYMENT_WITH_DOWN: 266.16,
  MSRP_PERCENTAGE: 1.31,
  PAYMENT_ZERO_DOWN_PRE_TAX: 300.99,
  TOTAL_LEASE_COST_ZERO_DOWN: 14538.2,
  TOTAL_LEASE_COST_WITH_DOWN: 14984.2,
};

export const DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_LEASE_PAYMENT = {
  ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
  taxMethod: TAX_ON_TOTAL_LEASE_PAYMENT,
};

export const DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_LEASE_PAYMENT = {
  ...DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
  taxMethod: TAX_ON_TOTAL_LEASE_PAYMENT,
};

export const WHEN_TAX_ON_LEASE_PAYMENT = {
  PAYMENT_ZERO_DOWN: 300.99,
  PAYMENT_WITH_DOWN: 266.16,
  MSRP_PERCENTAGE: 1.31,
  PAYMENT_ZERO_DOWN_PRE_TAX: 300.99,
  TOTAL_LEASE_COST_ZERO_DOWN: 12606.18,
  TOTAL_LEASE_COST_WITH_DOWN: 13222.86,
};


tests/LeaseCalculator.spec.js
import LeaseCalculator from "../src/LeaseCalculator";
import {
  // Tax applied on monthly payment
  DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
  DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
  WHEN_TAX_ON_MONTHLY_PAYMENT,
  // Tax applied on sales price
  DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_SALES_PRICE,
  DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_SALES_PRICE,
  WHEN_TAX_ON_SALES_PRICE,
  // Tax applied on total lease payment
  DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_LEASE_PAYMENT,
  DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_LEASE_PAYMENT,
  WHEN_TAX_ON_LEASE_PAYMENT,
  // RV
  DUMMY_LEASE_WITH_PERCENTAGE_RV,
  RV_VALUE,
  RV_PERCENTAGE,
  ACQUISITION_FEE_TOYOTA,
  DISPOSITION_FEE_TOYOTA,
} from "./constants";

describe("LeaseCalculator", () => {
  let leaseCalculator;
  beforeAll(() => {
    leaseCalculator = new LeaseCalculator();
  });

  describe("Validation", () => {
    it("should throw an error when missing a required field", () => {
      expect(() => {
        leaseCalculator.calculate({
          ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
          msrp: null,
        });
      }).toThrow(Error);
    });
    it("should throw an error when MSRP field is missing", () => {
      expect(() => {
        leaseCalculator.calculate({
          ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
          msrp: null,
        });
      }).toThrowError(`Invalid Input: MSRP`);
    });
    it("should throw an error when Selling Price field is missing", () => {
      expect(() => {
        leaseCalculator.calculate({
          ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
          sellingPrice: null,
        });
      }).toThrowError(`Invalid Input: Selling Price`);
    });
    it("should throw an error when RV field is missing", () => {
      expect(() => {
        leaseCalculator.calculate({
          ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
          rv: null,
        });
      }).toThrowError(`Invalid Input: Residual Value`);
    });
    it("should throw an error when MF field is missing", () => {
      expect(() => {
        leaseCalculator.calculate({
          ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
          mf: null,
        });
      }).toThrowError(`Invalid Input: Money Factor`);
    });
  });

  describe("Acquisition fee", () => {
    it("is 0 when Make is not specified", () => {
      leaseCalculator.calculate({
        ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
        make: "",
      });
      const acquisitionFee = leaseCalculator.getAcquisitionFee();
      expect(acquisitionFee).toEqual(0);
    });
    it("is based on Make, when specified", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const acquisitionFee = leaseCalculator.getAcquisitionFee();
      expect(acquisitionFee).toEqual(ACQUISITION_FEE_TOYOTA);
    });
  });

  describe("Disposition fee", () => {
    it("is 0 when Make is not specified", () => {
      leaseCalculator.calculate({
        ...DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT,
        make: "",
      });
      const acquisitionFee = leaseCalculator.getDispositionFee();
      expect(acquisitionFee).toEqual(0);
    });
    it("is based on Make, when specified", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const acquisitionFee = leaseCalculator.getDispositionFee();
      expect(acquisitionFee).toEqual(DISPOSITION_FEE_TOYOTA);
    });
  });

  describe("RV", () => {
    it("is the correct percentage off MSRP", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const rv = leaseCalculator.getRVPercentage();
      expect(rv).toEqual(RV_PERCENTAGE);
    });
    it("is the correct absolute value when passed as percentage", () => {
      leaseCalculator.calculate(DUMMY_LEASE_WITH_PERCENTAGE_RV);
      const rv = leaseCalculator.getRVValue();
      expect(rv).toEqual(RV_VALUE);
    });
    it("gets correct monthly payment when passed as percentage", () => {
      leaseCalculator.calculate(DUMMY_LEASE_WITH_PERCENTAGE_RV);
      const payment = leaseCalculator.getMonthlyPayment();
      expect(payment).toEqual(WHEN_TAX_ON_MONTHLY_PAYMENT.PAYMENT_ZERO_DOWN);
    });
  });

  describe("Tax on monthly payment", () => {
    it("gets correct monthly payment w/ $0 down", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const payment = leaseCalculator.getMonthlyPayment();
      expect(payment).toEqual(WHEN_TAX_ON_MONTHLY_PAYMENT.PAYMENT_ZERO_DOWN);
    });
    it("gets correct monthly payment w/ a down payment", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const payment = leaseCalculator.getMonthlyPayment();
      expect(payment).toEqual(WHEN_TAX_ON_MONTHLY_PAYMENT.PAYMENT_WITH_DOWN);
    });
    it("gets correct discount off msrp", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const offMsrp = leaseCalculator.getDiscountOffMsrpPercentage();
      expect(offMsrp).toEqual(WHEN_TAX_ON_MONTHLY_PAYMENT.OFF_MSRP);
    });
    it("gets correct MSRP percentage (1% rule)", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const msrpPercentage = leaseCalculator.getMonthlyPaymentToMsrpPercentage();
      expect(msrpPercentage).toEqual(
        WHEN_TAX_ON_MONTHLY_PAYMENT.MSRP_PERCENTAGE
      );
    });
    it("gets correct pre-tax monthly payment", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const paymentPreTax = leaseCalculator.getMonthlyPaymentPreTax();
      expect(paymentPreTax).toEqual(
        WHEN_TAX_ON_MONTHLY_PAYMENT.PAYMENT_ZERO_DOWN_PRE_TAX
      );
    });
    it("gets correct APR", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const apr = leaseCalculator.getAPR();
      expect(apr).toEqual(WHEN_TAX_ON_MONTHLY_PAYMENT.APR);
    });
    it("gets correct total lease cost w/ $0 down", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const totalCost = leaseCalculator.getTotalLeaseCost();
      expect(totalCost).toEqual(
        WHEN_TAX_ON_MONTHLY_PAYMENT.TOTAL_LEASE_COST_ZERO_DOWN
      );
    });
    it("gets correct total lease cost w/ a down payment", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_MONTHLY_PAYMENT
      );
      const totalCost = leaseCalculator.getTotalLeaseCost();
      expect(totalCost).toEqual(
        WHEN_TAX_ON_MONTHLY_PAYMENT.TOTAL_LEASE_COST_WITH_DOWN
      );
    });
  });

  describe("Tax on sales price", () => {
    it("gets correct monthly payment w/ $0 down", () => {
      leaseCalculator.calculate(DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_SALES_PRICE);
      const payment = leaseCalculator.getMonthlyPayment();
      expect(payment).toEqual(WHEN_TAX_ON_SALES_PRICE.PAYMENT_ZERO_DOWN);
    });

    it("gets correct monthly payment w/ a down payment", () => {
      leaseCalculator.calculate(DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_SALES_PRICE);
      const payment = leaseCalculator.getMonthlyPayment();
      expect(payment).toEqual(WHEN_TAX_ON_SALES_PRICE.PAYMENT_WITH_DOWN);
    });

    it("gets correct MSRP percentage (1% rule)", () => {
      leaseCalculator.calculate(DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_SALES_PRICE);
      const msrpPercentage = leaseCalculator.getMonthlyPaymentToMsrpPercentage();
      expect(msrpPercentage).toEqual(WHEN_TAX_ON_SALES_PRICE.MSRP_PERCENTAGE);
    });

    it("gets correct total lease cost w/ $0 down", () => {
      leaseCalculator.calculate(DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_SALES_PRICE);
      const totalCost = leaseCalculator.getTotalLeaseCost();
      expect(totalCost).toEqual(
        WHEN_TAX_ON_SALES_PRICE.TOTAL_LEASE_COST_ZERO_DOWN
      );
    });

    it("gets correct total lease cost w/ a down payment", () => {
      leaseCalculator.calculate(DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_SALES_PRICE);
      const totalCost = leaseCalculator.getTotalLeaseCost();
      expect(totalCost).toEqual(
        WHEN_TAX_ON_SALES_PRICE.TOTAL_LEASE_COST_WITH_DOWN
      );
    });
  });

  describe("Tax on total lease payment", () => {
    it("gets correct monthly payment w/ $0 down", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_LEASE_PAYMENT
      );
      const payment = leaseCalculator.getMonthlyPayment();
      expect(payment).toEqual(WHEN_TAX_ON_LEASE_PAYMENT.PAYMENT_ZERO_DOWN);
    });

    it("gets correct monthly payment w/ a down payment", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_LEASE_PAYMENT
      );
      const payment = leaseCalculator.getMonthlyPayment();
      expect(payment).toEqual(WHEN_TAX_ON_LEASE_PAYMENT.PAYMENT_WITH_DOWN);
    });

    it("gets correct MSRP percentage (1% rule)", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_LEASE_PAYMENT
      );
      const msrpPercentage = leaseCalculator.getMonthlyPaymentToMsrpPercentage();
      expect(msrpPercentage).toEqual(WHEN_TAX_ON_LEASE_PAYMENT.MSRP_PERCENTAGE);
    });

    it("gets correct total lease cost w/ $0 down", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_ZERO_DOWN_WITH_TAX_ON_LEASE_PAYMENT
      );
      const totalCost = leaseCalculator.getTotalLeaseCost();
      expect(totalCost).toEqual(
        WHEN_TAX_ON_LEASE_PAYMENT.TOTAL_LEASE_COST_ZERO_DOWN
      );
    });

    it("gets correct total lease cost w/ a down payment", () => {
      leaseCalculator.calculate(
        DUMMY_LEASE_WITH_DOWN_WITH_TAX_ON_LEASE_PAYMENT
      );
      const totalCost = leaseCalculator.getTotalLeaseCost();
      expect(totalCost).toEqual(
        WHEN_TAX_ON_LEASE_PAYMENT.TOTAL_LEASE_COST_WITH_DOWN
      );
    });
  });
});


src/constants.js
export const TAX_ON_MONTHLY_PAYMENT = 1;
export const TAX_ON_SALES_PRICE = 2;
export const TAX_ON_TOTAL_LEASE_PAYMENT = 3;

export const MAKES = [
  { id: 1, displayName: "Acura" },
  { id: 2, displayName: "Alfa Romeo" },
  { id: 3, displayName: "Audi" },
  { id: 4, displayName: "Bmw" },
  { id: 5, displayName: "Buick" },
  { id: 6, displayName: "Cadillac" },
  { id: 7, displayName: "Chevrolet" },
  { id: 8, displayName: "Chrysler" },
  { id: 9, displayName: "Dodge" },
  { id: 10, displayName: "Fiat" },
  { id: 11, displayName: "Ford" },
  { id: 12, displayName: "Genesis" },
  { id: 13, displayName: "GMC" },
  { id: 14, displayName: "Honda" },
  { id: 15, displayName: "Hyundai" },
  { id: 16, displayName: "Infiniti" },
  { id: 17, displayName: "Jaguar" },
  { id: 18, displayName: "Jeep" },
  { id: 19, displayName: "Kia" },
  { id: 20, displayName: "land rover" },
  { id: 21, displayName: "Lexus" },
  { id: 22, displayName: "Lincoln" },
  { id: 23, displayName: "Mini" },
  { id: 24, displayName: "Mazda" },
  { id: 25, displayName: "Mercedes benz" },
  { id: 26, displayName: "Mitsubishi" },
  { id: 27, displayName: "Nissan" },
  { id: 28, displayName: "Ram" },
  { id: 29, displayName: "Scion" },
  { id: 30, displayName: "Smart" },
  { id: 31, displayName: "Subaru" },
  { id: 32, displayName: "Tesla" },
  { id: 33, displayName: "Toyota" },
  { id: 34, displayName: "Volkswagen" },
  { id: 35, displayName: "Volvo" },
];
export const MANUFACTURER_FEES = [
  { makeId: 1, acquisitionFee: 595 },
  { makeId: 2, acquisitionFee: 595, dispositionFee: 395 },
  { makeId: 3, acquisitionFee: 895, dispositionFee: 495 },
  {
    makeId: 4,
    acquisitionFee: 925,
    dispositionFee: 350,
    mfReductionUnit: 0.00007,
  },
  { makeId: 5, acquisitionFee: 650, dispositionFee: 495 },
  { makeId: 6, acquisitionFee: 650, dispositionFee: 595 },
  { makeId: 7, acquisitionFee: 650, dispositionFee: 395 },
  { makeId: 8, acquisitionFee: 595, dispositionFee: 395 },
  { makeId: 9, acquisitionFee: 595, dispositionFee: 395 },
  { makeId: 10, acquisitionFee: 595, dispositionFee: 395 },
  { makeId: 11, acquisitionFee: 645, dispositionFee: 395 },
  { makeId: 12, acquisitionFee: 595, dispositionFee: 400 },
  { makeId: 13, acquisitionFee: 650, dispositionFee: 495 },
  { makeId: 14, acquisitionFee: 595, dispositionFee: 0 },
  { makeId: 15, acquisitionFee: 595, dispositionFee: 400 },
  {
    makeId: 16,
    acquisitionFee: 700,
    dispositionFee: 395,
    mfReductionUnit: 0.0001,
  },
  { makeId: 17, acquisitionFee: 895, dispositionFee: 300 },
  { makeId: 18, acquisitionFee: 795, dispositionFee: 395 },
  { makeId: 19, acquisitionFee: 595, dispositionFee: 400 },
  { makeId: 20, acquisitionFee: 895, dispositionFee: 300 },
  {
    makeId: 21,
    acquisitionFee: 795,
    dispositionFee: 350,
    mfReductionUnit: 0.00008,
  },
  { makeId: 22, acquisitionFee: 645, dispositionFee: 0 },
  { makeId: 23, acquisitionFee: 925, dispositionFee: 350 },
  { makeId: 24, acquisitionFee: 595, dispositionFee: 300 },
  {
    makeId: 25,
    acquisitionFee: 795,
    dispositionFee: 595,
    mfReductionUnit: 0.00007,
  },
  { makeId: 26, acquisitionFee: 595, dispositionFee: 0 },
  { makeId: 27, acquisitionFee: 650, dispositionFee: 395 },
  { makeId: 28, acquisitionFee: 595, dispositionFee: 395 },
  {
    makeId: 29,
    acquisitionFee: 650,
    dispositionFee: 350,
    mfReductionUnit: 0.00008,
  },
  { makeId: 30, acquisitionFee: 295, dispositionFee: 395 },
  { makeId: 31, acquisitionFee: 595, dispositionFee: 300 },
  { makeId: 32, acquisitionFee: 695, dispositionFee: 395 },
  {
    makeId: 33,
    acquisitionFee: 650,
    dispositionFee: 350,
    mfReductionUnit: 0.00008,
  },
  { makeId: 34, acquisitionFee: 675, dispositionFee: 395 },
  { makeId: 35, acquisitionFee: 995, dispositionFee: 350 },
];
